spring-webflux/src/main/java/org/springframework/web/reactive/socket/adapter/JettyWebSocketSession.java

```diff
@@ -51,52 +71,77 @@ public JettyWebSocketSession(Session session, HandshakeInfo info, DataBufferFact
 	public JettyWebSocketSession(Session session, HandshakeInfo info, DataBufferFactory factory,
 			@Nullable Sinks.Empty<Void> completionSink) {
 
-		super(session, ObjectUtils.getIdentityHexString(session), info, factory, completionSink);
-		// TODO: suspend causes failures if invoked at this stage
-		// suspendReceiving();
+		super(session, ObjectUtils.getIdentityHexString(session), info, factory);
+		this.handlerCompletionSink = completionSink;
+		this.flux = Flux.create(emitter -> {
+			this.sink = emitter;
+			emitter.onRequest(n -> {
+				boolean demand = false;
+				this.lock.lock();
+				try {
+					this.requested += n;
```

Comment:
this is more problematic, as `n` can be an "unbounded request" (`Integer.MAX_VALUE`) so `this.requested` can overflow to negative.

Reply:
```suggestion
					this.requested += n;
					if (this.requested < 0L) {
						this.requested = Long.MAX_VALUE;
					}
```

Reply:
Using `Math.addExact(long,long)` is probably best.

Reply:
I updated with the change suggested by @simonbasle not the `Math.addExact`. So once it reaches `Long.MAX_VALUE` it is considered an "unbounded request" and no longer decrements `this.requested` when demanding.